src/lib/project_manager/cocotb.ts

```diff
@@ -0,0 +1,299 @@
+            };
+            results.push(test_info);
+          }
+          resolve(results);
+        });
+      } catch (e) {
+        let results = this.get_all_test_fail(tests);
+        resolve(results);
+      }
+    });
+  };
+
+  get_all_test_fail(tests) {
+    let results: {}[] = [];
+    for (let i = 0; i < tests.length; i++) {
+      const test_name = tests[i];
+      let test_info = {
+        name: test_name,
+        pass: false
+      };
+      results.push(test_info);
+    }
+    return results;
+  }
+
+  async get_test_list(makefile_path: string) : Promise<TestItem[]> {
+    try {
+      shell.exec("make",(code, output) => {
+        if (code === 127){
+        vscode.window.showErrorMessage("Install Make to use cocotb");
+        }
+      });
+    } catch (error) {
+      vscode.window.showErrorMessage("Error testing deps."); 
+    }
+
+    let wksp_folder;
+    if(vscode.workspace.workspaceFolders !== undefined) {
+      wksp_folder = vscode.workspace.workspaceFolders[0].uri.fsPath;
+    } else {
+      return([]);
+    }
+
+    // let found_makefiles = glob.sync('/**/Makefile', {root: wksp_folder});
+    let found_makefiles = [makefile_path];
+    let cocotb_makefiles: {makefile: string, modules: [string]}[] = [];
+
+    const cocotb_makefile_must_contain = "include $(shell cocotb-config --makefiles)/Makefile.sim";
+
+    for (let item of found_makefiles) {
+      let data = fs.readFileSync(path_lib.resolve(item));
+      if (data.includes(cocotb_makefile_must_contain)) {
+        var new_data = data.toString().replace(cocotb_makefile_must_contain, '');
+        const tmpobj = tmp.fileSync();
+
+        fs.writeFileSync(tmpobj.name, `${new_data}\nprint-%  : ; @echo $* = $($*)\n`);
+
+        let modules = child_process.execSync(`make -f ${tmpobj.name} print-MODULE`).toString();
+        modules = modules.replace(/(\r\n|\n|\r)/, '');
+        let modules_arr = modules.split(' ');
+        modules_arr = modules_arr.filter(item => item !== "MODULE");
+        modules_arr = modules_arr.filter(item => item !== "=");
+        cocotb_makefiles.push({makefile: item, modules: modules_arr});
+      }
+    }
+
+    let test_array: TestItem[] = [];
+
+    for (let makefile of cocotb_makefiles) {
+      for (let module of makefile.modules) {
+        const test_name = module;
+        const module_path = path_lib.dirname(makefile.makefile);
+        const filename = path_lib.join(module_path, test_name);
+
+        let python_cocotb_data;
+        try {
+          python_cocotb_data = fs.readFileSync(`${filename}.py`);
+        } catch {
+          vscode.window.showErrorMessage(`Found "${module}" module in Makefile's MODULE variable but no "${test_name}.py" file found`);
+          continue;
+        }
+        
+        let input = python_cocotb_data.toString();
+        let regex = /(^[^#\r\n]* *@cocotb\.test[ \t]*\([ \t\w=.()"'\[\],]*\)[ \t]*(\r\n|\n|\r))([^#\r\n]*[ \t]*(async)?def[ \t]+([\w_]+)[ \t]*\([ \t\w_]+\)[ \t]*:[ \t]*$)/gm;
```

Comment:
The last thing to have all working (I guess):

I've found that after the change in the regex the parsed results of the tests don't get the right name. In my case `test_numer_1.undefined` is showed in the test list, hence the comparison here: https://github.com/TerosTechnology/vscode-terosHDL/blob/master/src/lib/project_manager/project_manager.ts#L804 is failing. 
Because of that the red and green icons in the list are not showed after running the tests.

So I guess that the challenge here is to be able find every cocotb test and get the same name that is in `results.xml` file.

Reply:
Seems to be working after testing it in a clean environment. I must have been missing something yesterday. My bad...